graphql-code-generator/src/main/java/io/ballerina/graphql/generator/service/combiner/ServiceTypesFileCombiner.java

```diff
@@ -0,0 +1,177 @@
+    private List<ModuleMemberDeclarationNode> inputObjectTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> interfaceTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> enumTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> unionTypesModuleMembers;
+    private List<ModuleMemberDeclarationNode> objectTypesModuleMembers;
+    private List<String> breakingChangeWarnings;
+
+    public ServiceTypesFileCombiner(ModulePartNode prevContentNode, ModulePartNode nextContentNode,
+                                    GraphQLSchema nextGraphqlSchema) {
+        this.prevContentNode = prevContentNode;
+        this.nextContentNode = nextContentNode;
+        this.nextGraphqlSchema = nextGraphqlSchema;
+
+        moduleMembers = new ArrayList<>();
+        inputObjectTypesModuleMembers = new ArrayList<>();
+        interfaceTypesModuleMembers = new ArrayList<>();
+        enumTypesModuleMembers = new ArrayList<>();
+        unionTypesModuleMembers = new ArrayList<>();
+        objectTypesModuleMembers = new ArrayList<>();
+
+        breakingChangeWarnings = new ArrayList<>();
+    }
+
+    public String generateMergedSrc() throws FormatterException {
+        String generatedSyntaxTree = Formatter.format(generateMergedSyntaxTree()).toString();
+        return Formatter.format(generatedSyntaxTree);
+    }
+
+    public SyntaxTree generateMergedSyntaxTree() {
+        List<ModuleMemberDeclarationNode> newMembers =
+                generateNewMembers(prevContentNode.members(), nextContentNode.members());
+
+        moduleMembers.addAll(inputObjectTypesModuleMembers);
+        moduleMembers.addAll(interfaceTypesModuleMembers);
+        moduleMembers.addAll(enumTypesModuleMembers);
+        moduleMembers.addAll(unionTypesModuleMembers);
+        moduleMembers.addAll(objectTypesModuleMembers);
+        moduleMembers.addAll(newMembers);
+
+        NodeList<ModuleMemberDeclarationNode> allMembers = createNodeList(moduleMembers);
+        ModulePartNode contentNode =
+                createModulePartNode(prevContentNode.imports(), allMembers, createToken(SyntaxKind.EOF_TOKEN));
+
+        TextDocument textDocument = TextDocuments.from(CodeGeneratorConstants.EMPTY_STRING);
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument);
+        return syntaxTree.modifyWith(contentNode);
+    }
+
+    private List<ModuleMemberDeclarationNode> generateNewMembers(NodeList<ModuleMemberDeclarationNode> prevMembers,
+                                                                 NodeList<ModuleMemberDeclarationNode> nextMembers) {
+        List<ModuleMemberDeclarationNode> newMembers = new ArrayList<>();
+        for (ModuleMemberDeclarationNode nextMember : nextMembers) {
+            boolean isFound = false;
+            for (ModuleMemberDeclarationNode prevMember : prevMembers) {
+                if (isMemberEquals(prevMember, nextMember)) {
+                    isFound = true;
```

Comment:
We can break the loop when we have a match, right? No need to iterate through the whole list.

```suggestion
                    isFound = true;
                    break;
```